crates/core/tedge_agent/src/software_update_manager/actor.rs

```diff
@@ -0,0 +1,193 @@
+use crate::software_update_manager::config::SoftwareUpdateManagerConfig;
+use crate::software_update_manager::error::SoftwareUpdateManagerError;
+use crate::software_update_manager::error::SoftwareUpdateManagerError::NoPlugins;
+use crate::state_repository::state::AgentStateRepository;
+use crate::state_repository::state::SoftwareOperationVariants;
+use crate::state_repository::state::State;
+use crate::state_repository::state::StateRepository;
+use crate::state_repository::state::StateStatus;
+use async_trait::async_trait;
+use plugin_sm::operation_logs::LogKind;
+use plugin_sm::operation_logs::OperationLogs;
+use plugin_sm::plugin_manager::ExternalPlugins;
+use plugin_sm::plugin_manager::Plugins;
+use std::sync::Arc;
+use tedge_actors::Actor;
+use tedge_actors::MessageReceiver;
+use tedge_actors::RuntimeError;
+use tedge_actors::Sender;
+use tedge_actors::SimpleMessageBox;
+use tedge_api::OperationStatus;
+use tedge_api::SoftwareRequestResponse;
+use tedge_api::SoftwareType;
+use tedge_api::SoftwareUpdateRequest;
+use tedge_api::SoftwareUpdateResponse;
+use tedge_config::ConfigRepository;
+use tedge_config::ConfigSettingAccessorStringExt;
+use tedge_config::SoftwarePluginDefaultSetting;
+use tedge_config::TEdgeConfigLocation;
+use tokio::sync::Mutex;
+use tracing::error;
+use tracing::log::warn;
+
+#[cfg(not(test))]
+const SUDO: &str = "sudo";
+#[cfg(test)]
+const SUDO: &str = "echo";
+
+pub struct SoftwareUpdateManagerActor {
+    config: SoftwareUpdateManagerConfig,
+    state_repository: AgentStateRepository,
+    operation_logs: OperationLogs,
+    message_box: SimpleMessageBox<SoftwareUpdateRequest, SoftwareUpdateResponse>,
+}
+
+#[async_trait]
+impl Actor for SoftwareUpdateManagerActor {
+    fn name(&self) -> &str {
+        "SoftwareUpdateManagerActor"
+    }
+
+    async fn run(&mut self) -> Result<(), RuntimeError> {
+        let plugins = Arc::new(Mutex::new(
```

Comment:
As for the `SoftwareListManagerActor`, using an `Arc<Mutex< >>` is useless as the `plugins` set is never used concurrently. 

This is also useless if these two actors are merged in a single one (as I strongly suggest).